Yank/pipeline.py

```diff
@@ -291,26 +263,92 @@ def prepare_phase(positions_file_path, topology_file_path, ligand_dsl, system_op
         raise RuntimeError(err_msg)
 
     # Create system and update box vectors (if needed)
-    system = topology_file.createSystem(removeCMMotion=False, **system_options)
+    system = parameters_file.createSystem(removeCMMotion=False, **system_options)
     if is_periodic:
         system.setDefaultPeriodicBoxVectors(*box_vectors)
 
+    return system
+
+
+def prepare_phase(positions_file_path, parameters_file_path, ligand_dsl, system_options,
+                  gromacs_include_dir=None, verbose=False):
+    """Create a Yank arguments for a phase from system files.
+
+    Parameters
+    ----------
+    positions_file_path : str
+        Path to system position file (e.g. 'complex.inpcrd/.gro/.pdb').
+    parameters_file_path : str
+        Path to system parameters file (e.g. 'complex.prmtop/.top/.xml').
+    ligand_dsl : str
+        MDTraj DSL string that specify the ligand atoms.
+    system_options : dict
+        system_options[phase] is a a dictionary containing options to
+        pass to createSystem(). If the parameters file is an OpenMM
+        system in XML format, this will be ignored.
+    gromacs_include_dir : str, optional
+        Path to directory in which to look for other files included
+        from the gromacs top file.
+    verbose : bool
+        Whether or not to log information (default is False).
+
+    Returns
+    -------
+    alchemical_phase : AlchemicalPhase
+        The alchemical phase for Yank calculation with unspecified name, and protocol.
+
+    """
+    # Load system files
+    if os.path.splitext(parameters_file_path)[1] == '.xml':
+        # Read Amber prmtop and inpcrd files
+        if verbose:
+            logger.info("xml: %s" % parameters_file_path)
+            logger.info("pdb: %s" % positions_file_path)
+        with open(parameters_file_path, 'r') as f:
+            serialized_system = f.read()
+        system = openmm.XmlSerializer.deserialize(serialized_system)
+        positions_file = openmm.app.PDBFile(positions_file_path)
+        parameters_file = positions_file  # needed for topology
+    else:
```

Comment:
The two branches are fundamentally different because with `.prmtop` and `.top` file we need to create the system (and thus build and pass system options, set box vectors etc), while with `.xml` files we just need to deserialize it.

The exception will be raised anyway because in `OpenMM` won't be able to parse the file if it has a different format, but I'll add a better error message.